nifi-extension-bundles/nifi-kafka-bundle/nifi-kafka-3-service/src/main/java/org/apache/nifi/kafka/service/Kafka3ConnectionService.java

```diff
@@ -0,0 +1,402 @@
+        }
+        if (producerConfiguration.getDeliveryGuarantee() != null) {
+            propertiesProducer.put(ProducerConfig.ACKS_CONFIG, producerConfiguration.getDeliveryGuarantee());
+        }
+        if (producerConfiguration.getCompressionCodec() != null) {
+            propertiesProducer.put(ProducerConfig.COMPRESSION_TYPE_CONFIG, producerConfiguration.getCompressionCodec());
+        }
+        final String partitionClass = producerConfiguration.getPartitionClass();
+        if (partitionClass != null && partitionClass.startsWith("org.apache.kafka")) {
+            propertiesProducer.put(ProducerConfig.PARTITIONER_CLASS_CONFIG, partitionClass);
+        }
+        return new Kafka3ProducerService(propertiesProducer, serviceConfiguration, producerConfiguration);
+    }
+
+    @Override
+    public List<ConfigVerificationResult> verify(final ConfigurationContext configurationContext, final ComponentLog verificationLogger, final Map<String, String> variables) {
+        final List<ConfigVerificationResult> results = new ArrayList<>();
+
+        final Properties clientProperties = getClientProperties(configurationContext);
+        clientProperties.putAll(variables);
+        try (final Admin admin = Admin.create(clientProperties)) {
+            final ListTopicsResult listTopicsResult = admin.listTopics();
+
+            final KafkaFuture<Collection<TopicListing>> requestedListings = listTopicsResult.listings();
+            final Collection<TopicListing> topicListings = requestedListings.get(VERIFY_TIMEOUT.toMillis(), TimeUnit.MILLISECONDS);
+            final String topicListingExplanation = String.format("Topics Found [%d]", topicListings.size());
+            results.add(
+                    new ConfigVerificationResult.Builder()
+                            .verificationStepName(TOPIC_LISTING_STEP)
+                            .outcome(SUCCESSFUL)
+                            .explanation(topicListingExplanation)
+                            .build()
+            );
+        } catch (final Exception e) {
+            verificationLogger.error("Kafka Broker verification failed", e);
+            results.add(
+                    new ConfigVerificationResult.Builder()
+                            .verificationStepName(CONNECTION_STEP)
+                            .outcome(FAILED)
+                            .explanation(e.getMessage())
+                            .build()
+            );
+        }
+
+        return results;
+    }
+
+    private Properties getConsumerProperties(final PropertyContext propertyContext, final Properties defaultProperties) {
+        final Properties properties = new Properties();
+        properties.putAll(defaultProperties);
+
+        // since config for ConsumerPool is locked in at ControllerService.enable(),
+        final boolean honorTransactions = propertyContext.getProperty(HONOR_TRANSACTIONS).asBoolean();
```

Comment:
Rather than continuing to keep the `Honor Transactions` or `Use Transactions` label, this seems like a good opportunity to reflect the Kafka property name and call it `Transaction Isolation Level`.